### PR TITLE
STIG update RHEL7 add gdm settings

### LINKED ISSUE #1160: need new rule: GDM AutomaticLoginEnable=false
image

### LINKED ISSUE #1161: need new rule: GDM TimedLoginEnable=false
image

### PR DESCRIPTION
**Pull request by redhatrises**
- Add new XCCDF/OVAL gdm settings content
- Fixes #1161
- Fixes #1160

### CODE REVIEW

**Comment by redhatrises** [comment]
Bump for review. Would like to get this in so that I can add remaining dconf changes and additional content.

File: RHEL/7/input/oval/gnome_gdm_disable_automatic_login.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+  comment="Disable GDM Automatic Login"
+  id="test_disable_automatic_login" version="1">
+    <ind:object object_ref="obj_disable_automatic_login" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="obj_disable_automatic_login"
+  version="1">
+    <ind:filepath>/etc/gdm/custom.conf</ind:filepath>
+    <ind:pattern operation="pattern match">^\[daemon]([^\n]*\n+)+?AutomaticLoginEnable=False$</ind:pattern>
```
Comment (shawndwells) [inline RHEL/7/input/oval/gnome_gdm_disable_automatic_login.xml]: [Ff]alse

File: RHEL/7/input/oval/gnome_gdm_disable_guest_login.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+  comment="Disable GDM Guest Login"
+  id="test_disable_guest_login" version="1">
+    <ind:object object_ref="obj_disable_guest_login" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="obj_disable_guest_login"
+  version="1">
+    <ind:filepath>/etc/gdm/custom.conf</ind:filepath>
+    <ind:pattern operation="pattern match">^\[daemon]([^\n]*\n+)+?TimedLoginEnable=False$</ind:pattern>
```
Comment (shawndwells) [inline RHEL/7/input/oval/gnome_gdm_disable_guest_login.xml]: [Ff]alse

File: RHEL/7/input/xccdf/system/software/gnome.xml [hunk 3]
```diff
@@ -12,7 +12,72 @@ Red Hat Graphical environment.
...
+<description>In the default GNOME3 desktop, the login is displayed after system boot
+and can display user accounts, allow users to reboot the system, and allow users to
+login automatically and/or with a guest account. The login screen should be configured
+to prevent such behavior.
+<br /><br />
+For more information about enforcing preferences in the GNOME3 environment using the DConf
+configuration system, see <b>http://wiki.gnome.org/dconf</b> and
+the man page <tt>dconf(1)</tt>. For Red Hat specific information on configuring DConf
+settings, see <b>https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Desktop_Migration_and_Administration_Guide/index.html</b>
+</description>
```
Comment (shawndwells) [inline RHEL/7/input/xccdf/system/software/gnome.xml]: I'm super leery of pointing to gnome.org. The upstream may have significantly different versions and config options than shipping in RHEL.  The other pointers make absolute sense!

File: RHEL/7/input/xccdf/system/software/gnome.xml [hunk 4]
```diff
@@ -12,7 +12,72 @@ Red Hat Graphical environment.
...
+user interaction or credentials. User should always be required to authenticate themselves
+to the system that they are authorized to use. To disable user ability to automatically
+login to the system, set the <tt>AutomaticLoginEnable</tt> to <tt>False</tt> in the
+<tt>[daemon]</tt> section in <tt>/etc/gdm/custom.conf</tt>. For example:
+<pre>[daemon]
+AutomaticLoginEnable=False</pre>
+</description>
+<ocil clause="GDM allows users to automatically login">
+To verify that automatic logins are disabled, run the following command:
+<pre>$ grep -Pzoi "^\[daemon]\\nautomaticlogin.*" /etc/gdm/custom.conf</pre>
```
Comment (shawndwells) [inline RHEL/7/input/xccdf/system/software/gnome.xml]: Can we verify with gconftool-2?

**Comment by shawndwells** [comment]
- Few nit picks on OVAL regex (ref to comments in direct patches)

- Should we be using gconftool-2 to set config options, vs editing in config file? 
e.g. 
`````

#Disable lock screen
gconftool-2 --type bool --set /apps/panel/global/disable_lock_screen true ;

#Disable screensaver idle activation
gconftool-2 --type bool --set /apps/gnome-screensaver/idle_activation_enabled false ;

#Disable screensaver lock
gconftool-2 --type bool --set /apps/gnome-screensaver/lock_enabled false ;
`````

File: RHEL/7/input/xccdf/system/software/gnome.xml (hunk 3 above)
Comment (redhatrises) [inline RHEL/7/input/xccdf/system/software/gnome.xml]: @shawndwells I will remove that in a follow up PR for dconf settings.

**Comment by redhatrises** [comment]
> - Should we be using gconftool-2 to set config options, vs editing in config file? 

Actually unfortunately no. gconftool-2 is used to set gconf policies which are set in `/etc/gconf/*`. gconftool-2 cannot actually edit `/etc/gdm/custom.conf`.

**Comment by redhatrises** [comment]
@shawndwells Fixed regex.

**Comment by shawndwells** [comment]
content looks good, pending jenkins

**Comment by redhatrises** [comment]
[retest this please]

**Comment by iankko** [comment]
LGTM. ACK. Thank you for the change.

**Pull request merged**